File: fetch-entries.py
#!/usr/bin/env python3
import os
import requests

SUPABASE_URL = os.environ.get('NEXT_PUBLIC_SUPABASE_URL')
SUPABASE_KEY = os.environ.get('NEXT_PUBLIC_SUPABASE_ANON_KEY')

headers = {
    'apikey': SUPABASE_KEY,
    'Authorization': f'Bearer {SUPABASE_KEY}',
    'Content-Type': 'application/json'
}

response = requests.get(
    f'{SUPABASE_URL}/rest/v1/diary_entries',
    headers=headers,
    params={'order': 'created_at.desc', 'limit': 10}
)

if response.status_code == 200:
    entries = response.json()
    print('=== Latest Entries ===')
    for i, entry in enumerate(entries[:10]):
        print(f'\n{i + 1}. "{entry.get("title", "No title")}"')
        print(f'   Date: {entry.get("entry_date", entry.get("created_at", "Unknown"))}')
        print(f'   Published: {entry.get("published", False)}')
        print(f'   Created: {entry.get("created_at", "Unknown")}')
else:
    print(f'Error: {response.status_code}')
    print(response.text)

File: fetch-entries.mjs
import { createClient } from '@supabase/supabase-js';

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL;
const supabaseKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY;

const supabase = createClient(supabaseUrl, supabaseKey);

async function getEntries() {
  const { data, error } = await supabase
    .from('entries')
    .select('id, title, content, published, created_at, entry_date')
    .order('created_at', { ascending: false })
    .limit(10);
  
  if (error) {
    console.error('Error:', error);
    return;
  }
  
  console.log('=== Latest Entries ===');
  data.forEach((entry, i) => {
    console.log(`\n${i + 1}. "${entry.title}"`);
    console.log(`   Date: ${entry.entry_date || entry.created_at}`);
    console.log(`   Published: ${entry.published}`);
    console.log(`   Created: ${entry.created_at}`);
  });
}

getEntries();
